webgui/InstallGUI.cmake

```
@@ -2,7 +2,38 @@ message("Install npm packages...")
 
 execute_process(
   COMMAND cp ${CC_PACKAGE} ${INSTALL_SCRIPTS_DIR}/package.json
+  WORKING_DIRECTORY ${INSTALL_SCRIPTS_DIR})
+
+execute_process(
   COMMAND npm install
   WORKING_DIRECTORY ${INSTALL_SCRIPTS_DIR})
 
-message("Installation of npm packages are finished.")
\ No newline at end of file
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E create_symlink dojo-util node_modules/util
+  WORKING_DIRECTORY ${INSTALL_SCRIPTS_DIR})
+
+message("Installation of npm packages are finished.")
+
+message("Initiating the Dojo Build System...")
+
+# Collect CodeCompass view JavaScript files
+set(DOJO_VIEWLIST "")
+file(GLOB _jsfiles RELATIVE "${INSTALL_SCRIPTS_DIR}" "${INSTALL_SCRIPTS_DIR}/codecompass/view/[^A-Z]*.js")
+foreach(_jsfile ${_jsfiles})
+  string(REGEX REPLACE "\\.js*$" "" _jsfile ${_jsfile})
+  list(APPEND DOJO_VIEWLIST "'${_jsfile}'")
+endforeach()
+list(JOIN DOJO_VIEWLIST  ", " DOJO_VIEWLIST)
+
+set(DOJO_OPTIMIZE "")       # optimization for non-layer modules
+set(DOJO_LAYEROPTIMIZE "")  # optimization for layer modules
```

I meant the comment as in not a code comment but a descriptive comment for the variable itself, saved by CMake.

```suggestion
set(DOJO_OPTIMIZE "" CACHE STRING "Optimisation method for non-layer Dojo modules")
set(DOJO_LAYEROPTIMIZE "" CACHE STRING "Optimisation method for layer-built Dojo modules")
```

---

While providing the description in a fancy way, this will make these variables cached by CMake and configurable by the user e.g. through `cmake-gui`. These variables are not for the user to configure, they are for internal usage only.

---

I missed that, sorry.